pkg/apis/v1/nodepool_validation.go

```diff
@@ -26,7 +26,41 @@ import (
 
 // RuntimeValidate will be used to validate any part of the CRD that can not be validated at CRD creation
 func (in *NodePool) RuntimeValidate(ctx context.Context) (errs error) {
-	errs = multierr.Combine(in.Spec.Template.validateLabels(), in.Spec.Template.Spec.validateTaints(), in.Spec.Template.Spec.validateRequirements(ctx), in.Spec.Template.validateRequirementsNodePoolKeyDoesNotExist())
+	errs = multierr.Combine(
+		in.Spec.Template.validateLabels(),
+		in.Spec.Template.Spec.validateTaints(),
+		in.Spec.Template.Spec.validateRequirements(ctx),
+		in.Spec.Template.validateRequirementsNodePoolKeyDoesNotExist(),
+		in.validateReplicasConstraints(),
+	)
+	return errs
+}
+
+// validateReplicasConstraints validates that when replicas is set, incompatible fields are not set
+func (in *NodePool) validateReplicasConstraints() (errs error) {
+	if in.Spec.Replicas == nil {
+		return nil
+	}
+
+	if in.Spec.Weight != nil {
```

Comment:
Can we have CEL validation for weights? That will make sure that this misconfiguration is caught at runtime